### ISSUE TITLE
[Bug] "The media could not be loaded, either because the server or network failed or because the format is not supported."

### ISSUE THREAD

**Issue by akasaka99** [iv-org/invidious#2260]
Hi-
1) I'm on a Mac (Mojave 10.14)
2) Using Firefox, Chrome, Brave & Safari
3) Tried all the invidious instances available
4) Removed adblockers as per some post recommendation.
But I keep on getting  80% of the time error message: 
"The media could not be loaded, either because the server or network failed or because the format is not supported."
and the the video wont play. Not sure what the issue is?

**Comment by unixfox** [comment]
Please post the video ID and the instance used.

**Comment by tomasodehnal** [comment]
I have the same issue on a private instance. Example video ID is `NkRkuI0ZgX0`. In the browser developer console, I can see repeating 403 requests to `https://r4---sn-n02xgoxufvg3-2gbl.googlevideo.com/videoplayback`. I'll be glad to help in identifying the cause.

**Comment by SamantazFox** [comment]
Please read the following FAQ question:
https://github.com/iv-org/documentation/blob/master/FAQ.md#q-the-media-could-not-be-loaded

Does any of the solutions listed here solve your issue?

**Comment by akasaka99** [comment]
No it did not solve the issue, the issue is more or less the same with any instance.
You can't watch 3 videos in without having buffering issues or error messages, so I gave up on invidious as it's not stable as it is right now.

**Comment by tomasodehnal** [comment]
@SamantazFox Thanks for the link. I'm not sure if I tried a video with the issue on a public instance, but I tried reloading the page multiple times and it never helped. As for DRM, today I tried to load like 50 videos, most of them music clips, but didn't find a single one that doesn't work (including the one I mentioned earlier).
I'm not sure how exactly Google distributes the load and if my private instance could be selectively blocked on some videos only, but today all works fine.

**Comment by SamantazFox** [comment]
> No it did not solve the issue, the issue is more or less the same with any instance.
> You can't watch 3 videos in without having buffering issues or error messages, so I gave up on invidious as it's not stable as it is right now.

@asddsaz Sorry to hear that. We're trying our best, but Google is making things difficult for us!

> @SamantazFox Thanks for the link. I'm not sure if I tried a video with the issue on a public instance, but I tried reloading the page multiple times and it never helped. As for DRM, today I tried to load like 50 videos, most of them music clips, but didn't find a single one that doesn't work (including the one I mentioned earlier).
> I'm not sure how exactly Google distributes the load and if my private instance could be selectively blocked on some videos only, but today all works fine.

@tomasodehnal You're welcome! In theory, the recent PRs should have fixed most of the "instance is likely blocked" issues, but there may be some other factors or edge cases we're not aware of.

**Comment by FelipeResende** [comment]
Hi, I had the same issue and the FAQ solutions didn't work for me. Looking into syslog, I saw that invidious crashed but without any information in the stack trace. Then, I rebuilt invidious without "--release" to see where it crashed, but the error didn't occur again and I could watch the video.

So, it might be a workaround, but I'm not sure what is the impact of running it without building with --release.

**Comment by SamantazFox** [comment]
> Hi, I had the same issue and the FAQ solutions didn't work for me. Looking into syslog, I saw that invidious crashed but without any information in the stack trace. Then, I rebuilt invidious without "--release" to see where it crashed, but the error didn't occur again and I could watch the video.
> 
> So, it might be a workaround, but I'm not sure what is the impact of running it without building with --release.

There is currently a bug that causes recurrent crashes of invidious. See #2471

**Comment by unixfox** [comment]
Closing in favor of https://github.com/iv-org/invidious/issues/602 and https://github.com/iv-org/invidious/issues/2850

This issue is likely going to be somewhat fixed in https://github.com/iv-org/invidious/pull/3011

**Comment by Turab** [comment]
I know it has been quite a time since this issue is being talked and that some fixes published, but I have experienced this today with the latest release and the problem seems to continue on some deployment scenarios. I haven't seen this mentioned before anywhere, so I am sharing my solution to this old discussion for the googler.

As you know, dash option is actually auto-proxying the video in the instance, disregarding "local: false" setting. So this means the video needs to be proxied from your domain. When I inspected with dev tools, I saw that invidious is trying to fetch the video from the instance on port 3000 which is the default container port of invidious. But this will cause "the media could not be loaded" error if you are behind a proxy (like I am).

The solution is simple. Just edit your container's environment variable which is named "INVIDIOUS_CONFIG" and add two settings: "external_port: 443" (or 80 if you are not using SSL) and "domain: your.instance.domain". Example:

`INVIDIOUS_CONFIG="channel_threads: 1\ncheck_tables: true\nfeed_threads: 1\nexternal_port: 443\ndomain: your.instance.domain"`

You can also edit your config.yml file to do it in a more readable format but I prefer ENV vars.

**Comment by unixfox** [comment]
That's exactly what is documented here: https://docs.invidious.io/installation/#post-install-configuration ....

**Comment by Turab** [comment]
Not actually. The options, what they are used for and the reverse proxy case is explained and well documented there BUT it nowhere mentions that what we are discussing here, "The media could not be loaded" error is in any way related to those settings and that it can be resolved by setting them.

Maybe you can say "well if you are behind a reverse proxy, it shouldn't be needed to mention it since you are already supposed to follow reverse proxy case instructions"; but then here is the thing: even without those settings present, invidious can be successfully deployed and it works!

Consider an average user deploying invidious without a problem. Then they test it and everything works. They can watch videos on their instance. They will never know they should have set reverse proxy specific settings. They might not even know they are behind a reverse proxy because standart deploying just works!

Then they decide to switch quality to dash. That's when they will see "The media could not be loaded" error and they won't know the cause and they will google the error string and won't see anyone mentioning this can be caused by lack of reverse proxy settings.

As you said, reverse proxy case is documented there but as you see, including this issue we are commenting under, none of the similar issues mention that this is caused by lack of reverse proxy settings. You have previous comments under this very issue trying to help the OP; but even you haven't told them that there is this possibility that this error could be caused by this.

**Comment by Shahin-rmz** [comment]
Hello, I got same problem and I can not watch videos on my instance.
any idea how to solve the problem?
Thanks

**Comment by unixfox** [comment]
https://github.com/iv-org/invidious/issues/4947
